apiserver/facades/agent/uniter/uniter.go

```
@@ -1495,14 +1495,17 @@ func (u UniterAPI) readOneUnitSettings(
 			// See LP1876097
 			return nil, apiservererrors.ErrPerm
 		}
-		settings, err = u.readLocalApplicationSettings(ctx, relUUID, tag, authTag.(names.UnitTag))
+		unitName, err := coreunit.NewName(authTag.Id())
```

**suggestion:** I think we should refactor a bit more to separate authorization and service access.

* I think that `readLocalApplicationSettings` shouldn't mix name & tags, and have the below signature:
```go
func (
	ctx context.Context,
	relUUID corerelation.UUID,
	appName string,
	unitName coreunit.Name,
) (map[string]string, error) 
```

It may requires to refactor a bit `ReadLocalApplicationSettings`, but we see that both function mention `LP1876097` but it may be a good opportunity to merge a bit both function which look very alike, and have only one reference to the bug.

---

The signature of `readLocalApplicationSettings` not mixing name & tags, isn't a sufficient reason to make this change. `coreunit.NewName` should ever fail here, but in case it did, I'd rather fail early before we start making service calls. 

The application tag is required for authorization, doing it in `readLocalApplicationSettings` allows it to be done in one place rather than making other methods longer duplicating code.

---

Fair enough.